### ISSUE TITLE
Extend the query language to support data use

### ISSUE THREAD

**Issue by mcupak** [ga4gh/beacon-team#60]
After adding consent codes and ADA-M, we should consider supporting them at the level of the query language, for example allow to filter for results that can be used for clinical purposes.

**Comment by antbro** [comment]
Absolutely! I have always argued that we could use these formal data use models to support resources [databases, datasets, biosamples,  subjects, etc] with respect to

a) communicating data use conditions
b) querying/discovery actions
c) automation of sharing actions (simple cases only)

Beacon can do (a) immediately, and (b) soon afterwards, as a general extension of our query scope and language. This latter item is something which we really must start some serious work on, but it will open a can of worms in terms ethics, governance and the question of when discovery becomes a form of sharing/access.

Cheers
Tony

 Professor Anthony J Brookes
 Department of Genetics
 University of Leicester
 University Road
 Leicester, LE1 7RH, UK
 Tel: +44 (0)116 2523401
 Mbl: +44 (0)777 0620396
 Fax: +44 (0)116 2523378
 mailto: ajb97@leicester.ac.uk<mailto:ajb97@leicester.ac.uk>
 --
 ** Where is the wisdom we have lost in knowledge?
 ** Where is the knowledge we have lost in information?
 ** T.S. Eliot
________________________________

**Comment by bakerdb** [comment]
FYI, the permission-management system that Genetic Alliance’s Platform for Engaging Everyone Responsibly (PEER) <http://www.geneticalliance.org/programs/biotrust/peer> system uses has such formal models in place that consider far more than data use.  The permission-management system is driven by an ontology that considers the interrelationships among a number of factors, including legal regulations, institutional policies, and individual permissions.  The technology is provided by Private Access, which has been building the system, and the ontology behind it, for nearly a decade.  So it’s quite mature at this point (though we all know that ontologies are living entities).

In full disclosure, I am a consultant to both Private Access and Genetic Alliance.  However, I wanted to make sure you knew that this capability currently exists and has been deployed by over 40 health advocacy groups.

-Dixie


Dixie B. Baker, Ph.D.
Senior Partner
Martin, Blanck and Associates
Office (Redondo Beach, CA):  310-791-9671
Mobile:  310-279-2579

On Jul 27, 2016, at 6:29 PM, antbro <notifications@github.com<mailto:notifications@github.com>> wrote:

Absolutely! I have always argued that we could use these formal data use models to support resources [databases, datasets, biosamples, subjects, etc] with respect to

a) communicating data use conditions
b) querying/discovery actions
c) automation of sharing actions (simple cases only)

Beacon can do (a) immediately, and (b) soon afterwards, as a general extension of our query scope and language. This latter item is something which we really must start some serious work on, but it will open a can of worms in terms ethics, governance and the question of when discovery becomes a form of sharing/access.

Cheers
Tony

Professor Anthony J Brookes
Department of Genetics
University of Leicester
University Road
Leicester, LE1 7RH, UK
Tel: +44 (0)116 2523401
Mbl: +44 (0)777 0620396
Fax: +44 (0)116 2523378
mailto: ajb97@leicester.ac.uk<mailto:ajb97@leicester.ac.uk><mailto:ajb97@leicester.ac.uk>
--
** Where is the wisdom we have lost in knowledge?
** Where is the knowledge we have lost in information?
** T.S. Eliot
________________________________

**Comment by antbro** [comment]
...and to add to this, with information provided by Dixie I looked at the PEER system in detail, and it is indeed very powerful and built for a particular use case. That said, it is also worth mentioning that ADA-M and (to a degree) Consent Codes also go considerably beyond 'Data Use Conditions'. We merely use that as a convenient label. But ADA-M, for example, actually covers areas of subject consent, researcher and institutional requirements, protection for sensitive groups, etc, relating to items as broad ranging as research and clinical domains of use, contracts, publication, timeline restrictions, data security, data withdrawal/destruction or return of any subject data, data linking, subject recontact, IP, reporting back, collaboration and fees. So the scope is similar to PEER, but it is designed to be more generally applicable which implies less detail and ontology control than a targeted system like PEER.

PEER and I have floated the idea of exploring interoperability on the level of discovery via the two systems, but time and funding are major obstacles.

Cheers
Tony

**Comment by bakerdb** [comment]
You're right in saying that PEER is a specific use case of privacy technology.   The privacy technology is Private Access.  John Mattison is quite familiar with Private Access, as is the CEO and founder, Robert Shelton.  I've brought both into this conversation.

- Dixie


Dixie Baker
Office: 310-791-9671
Cell: 310-279-2579


On Jul 28, 2016, at 6:00 AM, antbro <notifications@github.com<mailto:notifications@github.com>> wrote:

...and to add to this, with information provided by Dixie I looked at the PEER system in detail, and it is indeed very powerful and built for a particular use case. That said, it is also worth mentioning that ADA-M and (to a degree) Consent Codes also go considerably beyond 'Data Use Conditions'. We merely use that as a convenient label. But ADA-M, for example, actually covers areas of subject consent, researcher and institutional requirements, protection for sensitive groups, etc, relating to items as broad ranging as research and clinical domains of use, contracts, publication, timeline restrictions, data security, data withdrawal/destruction or return of any subject data, data linking, subject recontact, IP, reporting back, collaboration and fees. So the scope is similar to PEER, but it is designed to be more generally applicable which implies less detail and ontology control than a targeted system like PEER.

PEER and I have floated the idea of exploring interoperability on the level of discovery via the two systems, but time and funding are major obstacles.

Cheers
Tony

**Comment by jrambla** [comment]
+1 to add it to the query language.
Not sure about if v.0.4
